ibis/util.py

```diff
@@ -522,6 +522,26 @@ def gen_name(namespace: str) -> str:
     return f"ibis_{namespace}_{uid}"
 
 
+def gen_name_from_path(path: str | Path) -> str:
```

Comment:
I think the idea of baking **any** part of the path into an identifier is a very bad idea.

Lots of software has weird limitations around paths, that we should not bring into Ibis, like the thing you've already discovered about MySQL.

In particular, paths can be arbitrary byte sequences, they're not required to be UTF-8, Latin-1, anything in between, or anything in particular except a sequence of bytes.

Is this implementation prepared to handle that? If not, then we shouldn't do it, because it's adding a new set of unnecessary assumptions. If you need to keep track of what tables map to what files, DuckDB has an argument for that, or if you're not using DuckDB then it's better done upstream in the system that does the reading.

Reply:
I can write some code to handle the "random sequence of bytes". I think it would just replace those with the empty string, or an underscore, or something. I don't think the behavior here matters that much, as long as it doesn't error, because this should be so rare. I think this means that the implementation should be able to be fairly simple and maintainable.

If I come up with something not terrible would you consider it @cpcloud? I strongly still like the main idea of this PR.

Reply:
I really don't like the idea of replace sequences of bytes with anything.

What are we gaining by doing any of this, that makes it a worthwhile maintenance tradeoff? In other words, why should we bake in new assumptions about paths?

Right now, I don't see an improvement from all the effort here that justifies the maintenance burden.